internal/clients/kibana/alerting.go

```diff
@@ -0,0 +1,181 @@
+	for _, action := range rule.Actions {
+		actions = append(actions, alerting.ActionsInner{
+			Group:  &action.Group,
+			Id:     &action.ID,
+			Params: action.Params,
+		})
+	}
+
+	if isCreate {
+		reqModel := alerting.CreateRuleRequest{
+			Consumer:   rule.Consumer,
+			Actions:    actions,
+			Enabled:    rule.Enabled,
+			Name:       rule.Name,
+			NotifyWhen: (*alerting.NotifyWhen)(&rule.NotifyWhen),
+			Params:     rule.Params,
+			RuleTypeId: rule.RuleTypeID,
+			Schedule: alerting.Schedule{
+				Interval: &rule.Schedule.Interval,
+			},
+			Tags:     rule.Tags,
+			Throttle: *alerting.NewNullableString(rule.Throttle),
+		}
+		req := client.CreateRule(ctxWithAuth, rule.SpaceID, "").KbnXsrf("true").CreateRuleRequest(reqModel)
+		ruleRes, res, err := req.Execute()
+		if err != nil && res == nil {
+			return nil, diag.FromErr(err)
+		}
+
+		defer res.Body.Close()
+		return ruleResponseToModel(rule.SpaceID, ruleRes), utils.CheckHttpError(res, "Unabled to create alerting rule")
+	}
+
+	reqModel := alerting.UpdateRuleRequest{
+		Actions:    actions,
+		Name:       rule.Name,
+		NotifyWhen: (*alerting.NotifyWhen)(&rule.NotifyWhen),
+		Params:     rule.Params,
+		Schedule: alerting.Schedule{
+			Interval: &rule.Schedule.Interval,
+		},
+		Tags:     rule.Tags,
+		Throttle: *alerting.NewNullableString(rule.Throttle),
+	}
+	req := client.UpdateRule(ctxWithAuth, rule.ID, rule.SpaceID).KbnXsrf("true").UpdateRuleRequest(reqModel)
+	ruleRes, res, err := req.Execute()
+	if err != nil && res == nil {
+		return nil, diag.FromErr(err)
+	}
+
+	defer res.Body.Close()
+	if diags := utils.CheckHttpError(res, "Unable to update alerting rule"); diags.HasError() {
+		return nil, diag.FromErr(err)
+	}
+
+	shouldBeEnabled := rule.Enabled != nil && *rule.Enabled
+	if shouldBeEnabled && !ruleRes.Enabled {
+		res, err := client.EnableRule(ctxWithAuth, rule.ID, rule.SpaceID).KbnXsrf("true").Execute()
+		if err != nil && res == nil {
+			return nil, diag.FromErr(err)
+		}
+
+		if diags := utils.CheckHttpError(res, "Unable to enable alerting rule"); diags.HasError() {
+			return nil, diag.FromErr(err)
+		}
+	}
+
+	if !shouldBeEnabled && ruleRes.Enabled {
+		res, err := client.DisableRule(ctxWithAuth, rule.ID, rule.SpaceID).KbnXsrf("true").Execute()
+		if err != nil && res == nil {
+			return nil, diag.FromErr(err)
+		}
+
+		if diags := utils.CheckHttpError(res, "Unable to disable alerting rule"); diags.HasError() {
+			return nil, diag.FromErr(err)
+		}
+	}
+
+	return ruleResponseToModel(rule.SpaceID, ruleRes), diag.Diagnostics{}
+}
+
+func GetAlertingRule(ctx context.Context, apiClient *clients.ApiClient, id, spaceID string) (*models.AlertingRule, diag.Diagnostics) {
+	client, err := apiClient.GetAlertingClient()
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+
+	ctxWithAuth := apiClient.SetAlertingAuthContext(ctx)
+	req := client.GetRule(ctxWithAuth, id, spaceID)
+	ruleRes, res, err := req.Execute()
+	if err != nil && res == nil {
```

Comment:
Why do we check that `res == nil` here? Does it make sense to check that `err != nil` only?